src/main/java/mate/academy/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.repository.book;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.BookSearchParameters;
+import mate.academy.model.Book;
+import mate.academy.repository.SpecificationBuilder;
+import mate.academy.repository.SpecificationProviderManager;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParameters) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.titles() != null && searchParameters.titles().length > 0) {
+            spec = spec.and(bookSpecificationProviderManager
+                    .getSpecificationProvider("title")
```

Comment:
All magic numbers, strings, and symbols in your code should be constants.
[this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names)